sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -460,27 +485,45 @@ export class ReceiverImpl<ReceivedMessageT extends ReceivedMessage | ReceivedMes
    */
   private _isReceivingMessages(): boolean {
     if (
-      this._context.streamingReceiver &&
-      this._context.streamingReceiver.isOpen() &&
-      this._context.streamingReceiver.isReceivingMessages
+      this._streamingReceiver &&
+      this._streamingReceiver.isOpen() &&
+      this._streamingReceiver.isReceivingMessages
     ) {
       return true;
     }
     if (
-      this._context.batchingReceiver &&
-      this._context.batchingReceiver.isOpen() &&
-      this._context.batchingReceiver.isReceivingMessages
+      this._batchingReceiver &&
+      this._batchingReceiver.isOpen() &&
+      this._batchingReceiver.isReceivingMessages
     ) {
       return true;
     }
     return false;
   }
 
   private _createBatchingReceiver(
-    context: ClientEntityContext,
+    context: ConnectionContext,
+    entityPath: string,
     options?: ReceiveOptions
   ): BatchingReceiver {
-    return BatchingReceiver.create(context, options);
+    return BatchingReceiver.create(context, entityPath, options);
+  }
+
+  /**
+   * Helper function to retrieve active receiver name, if it exists.
+   */
+  private _getAssociatedReceiverName(): string | undefined {
+    if (this._streamingReceiver && this._streamingReceiver.isOpen()) {
```

Comment:
Can you add in a comment that explains why one would take precedence over the other?

Reply:
the order does not matter, the service only needs the name of any active receiver link. Added a comment to reflect that in https://github.com/Azure/azure-sdk-for-js/pull/9710/commits/c6263653f6e469812aa986667eaab9e222ebfe3a.